GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -759,27 +759,43 @@ private void RegisterPlugins()
                 var pluginEntries = PluginRegistry.Plugins
                     .OrderBy(entry => entry.Name, StringComparer.CurrentCultureIgnoreCase);
 
+                // the menu already contains 2 items:
+                //    [0] Separator
+                //    [1] Plugin Settings
+                // insert plugins above the separator (the list is in reverse order)
+                int insertOffsetFromBelow = 2;
```

Comment:
Yes, this is clearer but still needs:
```suggestion
                const string PluginManager = "Plugin Manager";
                int insertOffsetFromBelow = existingPluginMenus.Contains(PluginManager) ? 3 : 2;
```

Reply:
Thinking of it, if we invert the order of the list and so add each plugin at the top (index '0'), the changes are even lighter/clearer.

```csharp
            lock (PluginRegistry.Plugins)
            {
                var pluginEntries = PluginRegistry.Plugins
                    .OrderByDescending(entry => entry.Name, StringComparer.CurrentCultureIgnoreCase);

                // the menu already contains 2 items:
                //    [0] Separator
                //    [1] Plugin Settings
                // insert plugins above the separator (the list is in reverse order)

                foreach (var plugin in pluginEntries)
                {
                    // don't add the plugin to the Plugins menu, if already added
                    if (existingPluginMenus.Contains(plugin))
                    {
                        continue;
                    }

                    var item = new ToolStripMenuItem
                    {
                        Text = plugin.Description,
                        Image = plugin.Icon,
                        Tag = plugin
                    };
                    item.Click += delegate
                    {
                        if (plugin.Execute(new GitUIEventArgs(this, UICommands)))
                        {
                            RefreshRevisions();
                        }
                    };

                    if (plugin.Name == PluginManagerName)
                    {
                        // insert Plugin Manager below the separator
                        pluginsToolStripMenuItem.DropDownItems.Insert(pluginsToolStripMenuItem.DropDownItems.Count - 1, item);
                    }
                    else
                    {
                        pluginsToolStripMenuItem.DropDownItems.Insert(0, item);
                    }
                }
            }
```

Do I miss something? Do you see a bad side effect?

Reply:
Yes, this should work and is clearer. And that _comment_ has become true again.
Just remove the blank line between the _comment_ and the `foreach`.